docs/protocol/avs.mdx

```diff
@@ -0,0 +1,246 @@
+
+:::note
+
+The IRemoteChallenger implementations are not yet live in production.
+
+:::
+
+### Workflow for registering
+
+Prerequisite: operator has to be registered as an Eigenlayer Operator (through their CLI)
+
+```mermaid
+sequenceDiagram
+  box Ethereum L1
+    actor Operator
+    participant ECDSAStakeRegistry
+    participant HSM
+    participant AVSDirectory
+  end
+
+Operator->>ECDSAStakeRegistry: registerOperatorWithSignature(operator,operatorSignature)
+ECDSAStakeRegistry->>HSM: registerOperatorToAVS(operator,operatorSignature)
+HSM->>AVSDirectory: registerOperatorToAVS(operator,operatorSignature)
+Operator->>HSM: enrollForChallengers(challenger[])
+```
+
+We need operators to enroll into specific challengers to allow for permissionless slashing by any ISM deployers. As an operator, you can inspect every remote challenger and choose for yourself which ones you want to opt into depending on risk vs reward. This also means Abacus Works won't be the bottleneck for adding challengers for different rollup stacks. The operators will be able to unenroll after the unenrollment delay blocks has passed.
+
+### Workflow for deregistering
+
+```mermaid
+sequenceDiagram
+  box Ethereum L1
+    actor Operator
+    participant HSM
+    participant IRemoteChallenger
+    participant ECDSAStakeRegistry
+    participant AVSDirectory
+  end
+
+Operator->>HSM: startUnenrollment(challenger[])
+Note over Operator,HSM: Operator waits for delay period
+Operator->>ECDSAStakeRegistry: deregisterOperatorWithSignature(operator)
+ECDSAStakeRegistry->>HSM: deregisterOperatorToAVS(operator)
+Operator->>HSM: completeUnenrollment(challenger[])
+HSM->>IRemoteChallenger: challengeDelayBlocks()
+HSM->>AVSDirectory: deregisterOperatorToAVS(operator)
+```
+
+Operators can only deregister themselves after unenrolling from all challengers they are enrolled in, each of which can have its own delay period. This is to ensure that the operator is not able to withdraw their stake before the challenge period has passed for any of the challengers.
+
+### Workflow for staking
+
+```mermaid
+sequenceDiagram
+  box Ethereum L1
+    actor Staker
+    participant DelegationManager
+    participant StrategyManager
+    participant Strategy1
+
+  end
+
+Staker->>DelegationManager: delegateTo(operator,approverSignatureAndExpiry)
+Staker->>StrategyManager: depositIntoStrategy(strategy,token,amt)
+StrategyManager->>Strategy1: token.transferFrom(sender, strategy, amount)
+StrategyManager->>DelegationManager: increaseDelegatedShares
+```
+
+In Eigenlayer's design, staker funds are safeguarded as they are not made accessible to the AVS until slashing is activated. This ensures that no funds are at risk prematurely. Additionally, this design does not allow users to selectively opt into the Hyperlane AVS alone, as staking is managed through the StrategyManager and directed towards a specific operator.
+
+### Workflow for reading stake from an ISM
```

Comment:
I would either remove this section for now, or move it to the bottom or a subpage